tests/test_retro_scan.py

```
@@ -265,6 +326,44 @@ def fake_discover(args):
+        with tempfile.TemporaryDirectory() as tmp:
+            transcript = Path(tmp) / "session.jsonl"
+            transcript.write_text('{"text":"opened PR #105"}\n', encoding="utf-8")
+
+            def fake_scan_prs(pr_numbers, repo_arg):
+                return {"repo": "o/r", "prs": {"105": []},
+                        "branches": {"105": None}}
+
+            argv = ["retro_scan.py", "--transcript", str(transcript), "--pr", "105"]
+            with mock.patch.object(retro_scan, "scan_prs", fake_scan_prs), \
+                    mock.patch.object(sys, "argv", argv):
+                with self.assertRaises(AttributeError):
```

Ruff (0.16.2)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)